core/server/api/notifications.js

```diff
@@ -83,7 +112,23 @@ notifications = {
      * @returns {Promise(Notifications)}
      */
     destroy: function destroy(options) {
-        return canThis(options.context).destroy.notification().then(function () {
+        var tasks;
+
+        /**
+         * ### Handle Permissions
+         * We need to be an authorised user to perform this action
+         * @param {Object} options
+         * @returns {Object} options
+         */
+        function handlePermissions(options) {
+            return canThis(options.context).destroy.notification().then(function () {
+                return options;
+            }, function () {
+                return Promise.reject(new errors.NoPermissionError('You do not have permission to destroy notifications.'));
+            });
+        }
+
+        function destroyNotification(options) {
```

Comment:
@ErisDS even before I moved this to pipeline I noticed that when a notification is deleted it does not decrement the notifications counter. Is this intended behavior or something that should be fixed?